From 62ef5bb1bde45671c5d6f22bd05a668f467e4fd9 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Wed, 22 Nov 2017 03:16:21 +0800
Subject: [PATCH 2877/5242] MLK-16919-1 ARM: dts: add a new property to
 indicate mipi mode for csi

commit  77f67029b97e061ce6fd3287e46f0819c79ed0c1 from
https://source.codeaurora.org/external/imx/linux-imx.git

in previous version, the property 'csi-mux-mipi' is used to set GPR for MIPI
mode, but on some SoC like iMX8MQ (reuse the CSI) there's no GPR MUX setting
for it.
Add a new property 'fsl,mipi-mode' and check this first for the mode.
If it's MIPI mode, then parse 'csi-mux-mipi'.

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Reviewed-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-sdb.dts |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 790f56f..4bdf1d4 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -256,6 +256,7 @@
 
 &csi1 {
 	csi-mux-mipi = <&gpr 0x14 4>;
+	fsl,mipi-mode;
 	status = "okay";
 
 	port {
-- 
1.7.9.5

